### PR TITLE
Adding nli_tr dataset

### PR DESCRIPTION
**Pull request by e-budur**
Hello,

In this pull request, we have implemented the necessary interface to add our recent dataset [NLI-TR](https://github.com/boun-tabi/NLI-TR).  The datasets will be presented on a full paper at EMNLP 2020 this month. [[arXiv link] ](https://arxiv.org/pdf/2004.14963.pdf)

The dataset is the neural machine translation of SNLI and MultiNLI datasets into Turkish.  So, we followed a similar format with the original datasets hosted in the HuggingFace datasets hub.  

Our dataset is designed to be accessed as follows by following the interface of the GLUE dataset that provides multiple datasets in a single interface over the HuggingFace datasets hub.  

```
from datasets import load_dataset
multinli_tr = load_dataset("nli_tr", "multinli_tr")
snli_tr = load_dataset("nli_tr", "snli_tr")
```

Thanks for your help in reviewing our pull request.

### CODE REVIEW

File: datasets/nli_tr/nli_tr.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+class NLITRConfig(datasets.BuilderConfig):
+    """ BuilderConfig for NLI-TR"""
+
+    def __init__(self, version=None, data_url=None, data_dir=None, **kwargs):
+        super(NLITRConfig, self).__init__(version=datasets.Version(version, ""), **kwargs)
+        self.data_url = data_url
+        self.data_dir = data_dir
+        self.description = _DESCRIPTION
+        self.homepage = _HOMEPAGE
+        self.citation = _CITATION
```
Comment (lhoestq) [inline datasets/nli_tr/nli_tr.py:57]: It looks like you only need the `data_url` attribute.
Could you remove `data_dir` and maybe use `_DESCRIPTION`, `_HOMEPAGE`, and `_CITATION` directly in the `_info()` method ?

File: datasets/nli_tr/nli_tr.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+
+
+class NliTr(datasets.GeneratorBasedBuilder):
+    """NLI-TR: The Turkish translation of SNLI and MultiNLI datasets using Amazon Translate."""
+
+    VERSION = datasets.Version("1.0.0")
+
```
Comment (lhoestq) [inline datasets/nli_tr/nli_tr.py:64]:
```suggestion
    BUILDER_CONFIG_CLASS = NLITRConfig
```
We need to specify here that a custom config is available ;)

**Review by lhoestq — COMMENTED** [review]
Looks good to me !
Thanks for adding this one ;)

I left minor comments. Once they're resolved we can merge it :)

File: datasets/nli_tr/nli_tr.py (hunk 1 above)
Comment (e-budur) [reply datasets/nli_tr/nli_tr.py]: Thank you @lhoestq for the time you take to review our pull request.  We've made the changes you described.  Hope that it is ready for being merged.  Please let me know if you have any additional requests for revisions.

**Comment by e-budur** [comment]
Thank you @lhoestq for the time you take to review our pull request. We appreciate your help.

We've made the changes you described. Hope that it is ready for being merged. Please let me know if you have any additional requests for revisions.

File: datasets/nli_tr/nli_tr.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+                    publisher = "Association for Computational Linguistics",
+                    abstract = "Large annotated datasets in NLP are overwhelmingly in English. This is an obstacle to progress in other languages. Unfortunately, obtaining new annotated resources for each task in each language would be prohibitively expensive. At the same time, commercial machine translation systems are now robust. Can we leverage these systems to translate English-language datasets automatically? In this paper, we offer a positive response for natural language inference (NLI) in Turkish. We translated two large English NLI datasets into Turkish and had a team of experts validate their translation quality and fidelity to the original labels. Using these datasets, we address core issues of representation for Turkish NLI. We find that in-language embeddings are essential and that morphological parsing can be avoided where the training set is large. Finally, we show that models trained on our machine-translated datasets are successful on human-translated evaluation sets. We share all code, models, and data publicly.",
+                }
+                """
```
Comment (lhoestq) [inline datasets/nli_tr/nli_tr.py:47]: Sorry maybe I wasn't clear about description, homepage and citation.
To appear in the datasets hub page on huggingface.co those three fields actually need to be in global variables _DESCRIPTION, _CITATION and _HOMEPAGE.
As they're global variables you don't need to have them in NLITRConfig, and you can directly use those three variables in _info().
See [squad.py](https://github.com/huggingface/datasets/blob/master/datasets/squad/squad.py) for example.

Comment (e-budur) [reply datasets/nli_tr/nli_tr.py]: Very sorry for the confusion. I've revised the implementation accordingly. Thank you so much for the time you take for reviewing it.

File: datasets/nli_tr/nli_tr.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+The Natural Language Inference in Turkish (NLI-TR) is a set of two large scale datasets that were obtained by translating the foundational NLI corpora (SNLI and MNLI) using Amazon Translate.
+"""
+
+_HOMEPAGE = "https://github.com/boun-tabi/NLI-TR"
+
+
+class NLITRConfig(datasets.BuilderConfig):
+    """ BuilderConfig for NLI-TR"""
+
+    def __init__(self, version=None, data_url=None, data_dir=None, **kwargs):
```
Comment (lhoestq) [inline datasets/nli_tr/nli_tr.py:51]:
```suggestion
    def __init__(self, version=None, data_url=None, **kwargs):
```
data_dir is not used here

**Review by lhoestq — APPROVED** [review]
Looks all good now thanks :)

**Pull request merged**